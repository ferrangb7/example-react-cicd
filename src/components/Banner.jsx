import { useNavigate } from "react-router-dom";
import { newReliclogs } from "../api/newReliclogs";

export const Banner = () => {

    // run tests

    const navigate = useNavigate();

    const handleClick = () => {
        (async () => {
            await newReliclogs('Start Button have been clicked!')
        })();
        navigate("/search")
    }

    return (
        <section className="banner">
            <div className="inner">
                <h1>THIS IS GEODISCOVER</h1>
                <p>
                    Discover the world like never before with GeoDiscover! Our intuitive web app provides detailed insights and captivating
                    images of countries worldwide, all at your fingertips. Unravel the hidden gems, explore diverse cultures, 
                    and embark on a virtual journey across the globe with GeoDiscover.
                </p>
            </div>
            <button onClick={ handleClick } type="button" className="btn btn-light">Start Now!</button>
        </section>
    );
}
