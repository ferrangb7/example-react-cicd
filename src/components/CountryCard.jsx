
import { useEffect, useState } from "react"
import { getInfo } from "../helpers/getInfo"
import axios from "axios";

export const CountryCard = ({ cty }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        const getImage = async ( countryName ) => {
            try {
                const { data } = await axios.get('https://api.pexels.com/v1/search', {
                    headers: {
                        Authorization: `${import.meta.env.VITE_API_KEY}`
                    },
                    params: {
                        query: countryName,
                        per_page: 5,
                        page: 1,
                    }
                })
                setImages(data.photos)
            } catch (err) {
                console.error(err);
            }
        }
        getImage(cty.name.common)
    }, [])
    

    return (
        <div className="card country">
            <img src={ cty.flags.svg } className="card-img-top" alt="..." />
            <div className="card-body">
                  <h5 className="card-title"> { cty.name.common } </h5>
                  <p className="card-text">{ cty.flags.alt }</p>
            </div>
            <ul className="list-group list-group-flush">
              <li className="list-group-item">Population: { cty.population }</li>
              <li className="list-group-item">Language(s): { getInfo(cty.languages) }</li>
              <li className="list-group-item">Currency: { getInfo(cty.currencies) }</li>
            </ul>
            <div className="card-body">
                <div className="card-group">
                {
                    images.map(img => (
                        <div className="card" key={img.id}><img src={ img.src.tiny } className="card-img-top" alt={ img.photographer }></img></div>
                    ))
                }
                </div>
            </div>
        </div>
    )
}
