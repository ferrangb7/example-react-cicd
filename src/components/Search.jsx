import { useState } from "react"
import { CountryCard } from "./CountryCard";
import { getCountryByName } from "../api/getCountryByName";
import { Error } from "./Error";
import { newReliclogs } from "../api/newReliclogs";
import CheckboxWithLabel from "./CheckboxWithLabel";

export const Search = () => {

    const [ countries, setCountries ] = useState([]);
    const [ inputValue, setInputValue ] = useState("");

    const onInputChange = (e) => {
        e.preventDefault();
        setInputValue(e.target.value);
    };

    const onSearchSubmit = async (e) => {
        e.preventDefault();
        const data = await getCountryByName(inputValue);
        setCountries(data);
        setInputValue('');
        data.forEach( async cty => {
            await newReliclogs(`Country Searched: ${ cty.name.common }`);
        });
    };    

    return(
        <>
            <div className="row search">
                <div className="col-3">
                    <h4>Searching</h4>
                    <hr />
                    <form onSubmit={ onSearchSubmit } aria-label="form">
                        <input
                            type="text"
                            placeholder="Search a country"
                            className="form-control"
                            autoComplete="off"
                            value={ inputValue }
                            onChange={ onInputChange }

                        />
                        <button className="btn btn-outline-primary mt-2">
                            search
                        </button>
                    </form>
                </div>
                <div className="col-9">

                    <h4>Results</h4>
                    <hr />
                    {
                        countries ? countries.map(cty => (
                            <CountryCard key={ cty.population } cty={ cty } />
                        )) : <Error />
                    }

                    <CheckboxWithLabel labelOn="on" labelOff="off" />
                </div>
            </div>
        </>
    )
}
