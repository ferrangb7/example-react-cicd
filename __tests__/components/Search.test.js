import renderer from 'react-test-renderer'; //you will need this for the snapshot test
import { cleanup, fireEvent, render } from '@testing-library/react'; //this is for the DOM testing
import { screen } from '@testing-library/dom';//this is for the DOM testing
import { Search } from '../../src/components/Search';

/*
4. This test suite is all about snapshot testing! Testing the UI can be hard, but we can use snapshot testing
to make sure that nothing gets moved around or changed unexpectedly.
*/

//TODO: write a snapshot test that checks that the Search.js component actually renders, then change something to see the test fail
//Look at this doc to help you write the test: https://jestjs.io/docs/snapshot-testing

test("renders correctly", () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test("renders correctly", () => {
//   const tree = renderer.create(<Search />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

