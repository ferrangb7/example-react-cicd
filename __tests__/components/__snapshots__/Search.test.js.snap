// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row search"
>
  <div
    className="col-3"
  >
    <h4>
      Searching
    </h4>
    <hr />
    <form
      aria-label="form"
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control"
        onChange={[Function]}
        placeholder="Search a country"
        type="text"
        value=""
      />
      <button
        className="btn btn-outline-primary mt-2"
      >
        search
      </button>
    </form>
  </div>
  <div
    className="col-9"
  >
    <h4>
      Results
    </h4>
    <hr />
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      off
    </label>
  </div>
</div>
`;
